lib/src/Uncertainty/Algorithm/Sensitivity/RankSobolSensitivityAlgorithm.cxx

```diff
@@ -0,0 +1,419 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief Implementation for Rank obol Sensitivity Algorithm
+ *
+ *  Copyright 2005-2024 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "openturns/RankSobolSensitivityAlgorithm.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/Cloud.hxx"
+#include "openturns/Text.hxx"
+#include "openturns/KernelSmoothing.hxx"
+#include "openturns/TBBImplementation.hxx"
+#include "openturns/RandomGenerator.hxx"
+#include "openturns/KPermutationsDistribution.hxx"
+#include "openturns/Curve.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(RankSobolSensitivityAlgorithm)
+
+static const Factory<RankSobolSensitivityAlgorithm> Factory_RankSobolSensitivityAlgorithm;
+
+/* Default constructor */
+RankSobolSensitivityAlgorithm::RankSobolSensitivityAlgorithm()
+  : SobolIndicesAlgorithmImplementation()
+{
+  // Nothing to do
+}
+
+/** Constructor with parameters */
+RankSobolSensitivityAlgorithm::RankSobolSensitivityAlgorithm(const Sample & inputDesign,
+    const Sample & outputDesign)
```

Comment:
you need to call the parent class ctor, at least ": SobolIndicesAlgorithmImplementation()"